youtube_dl/extractor/douyutv.py

```diff
@@ -75,74 +73,74 @@ def _real_extract(self, url):
             room_id = self._html_search_regex(
                 r'"room_id"\s*:\s*(\d+),', page, 'room id')
 
-        config = None
+        room_url = 'http://m.douyu.com/html5/live?roomId=%s' % room_id
+        room_content = self._download_webpage(room_url, video_id)
+        room_json = self._parse_json(room_content, video_id, fatal=False)
+
+        room = room_json['data']
+
+        show_status = room.get('show_status')
+        # 1 = live, 2 = offline
+        if show_status == '2':
+            raise ExtractorError(
+                'Live stream is offline', expected=True)
+
+        flv_json = None
         # Douyu API sometimes returns error "Unable to load the requested class: eticket_redis_cache"
         # Retry with different parameters - same parameters cause same errors
         for i in range(5):
-            prefix = 'room/%s?aid=android&client_sys=android&time=%d' % (
-                room_id, int(time.time()))
-            auth = hashlib.md5((prefix + '1231').encode('ascii')).hexdigest()
+            tt = int(time.time() / 60)
+            did = uuid.uuid4().hex.upper()
+
+            sign_content = '{room_id}{did}A12Svb&%1UUmf@hC{tt}'.format(room_id = room_id, did = did, tt = tt)
+            sign = hashlib.md5((sign_content).encode('utf-8')).hexdigest()
+
+            payload = {'cdn': 'ws', 'rate': '0', 'tt': tt, 'did': did, 'sign': sign}
+            flv_data = compat_urllib_parse_urlencode(payload)
 
-            config_page = self._download_webpage(
-                'http://www.douyutv.com/api/v1/%s&auth=%s' % (prefix, auth),
-                video_id)
+            flv_request_url = 'http://www.douyu.com/lapi/live/getPlay/%s' % room_id
+            flv_request = sanitized_Request(flv_request_url, flv_data,
```

Comment:
When I use ```_download_webpage``` with ```query``` parameter , it always returns 403 error.
I don't know how to solve, so I use ```sanitized_Request``` that is more easier to me.